### PR TITLE
Add func name

### PR DESCRIPTION
**Pull request by PracticalMetal**
Hi, I made the following changes: Introduced string s which will store __PRETTY_FUNC__ and made friend class to access that data. I am not sure if this fixes the issue so I will be marking this PR as a draft, If this doesn't fix the issue or is not how you intend it to be please let me know what I did wrong and what I can improve and I will correct it. Thankyou!

### CODE REVIEW

**Comment by Dante-666** [comment]
Hey, this does not address the issue. I think the requirement is not clear. Let's take a step back.

Have you built the application locally and tried to run? Please try to follow the readme and run it in debug mode, make -j config=debug and observe the logs in the console.

You need to install curl with headers and another software called feh to get it to work.

File: src/logutil/logger.cc [hunk 1]
```diff
@@ -42,15 +42,19 @@ void Logger::log(const LogLevel &level, const string &message) {
...
 void Logger::LogDebug(const string &message) {
+  s=__PRETTY_FUNCTION__;
   Logger::log(LogLevel::DEBUG, message);
 }
 void Logger::LogInfo(const string &message) {
+  s=__PRETTY_FUNCTION__;
   Logger::log(LogLevel::INFO, message);
 }
 void Logger::LogWarn(const string &message) {
+  s=__PRETTY_FUNCTION__;
```
Comment (Dante-666) [inline src/logutil/logger.cc:53]: If you run this here, the _PRETTY_FUNCTION__ will tell you that this is Logger::LogWarn. This is not what we want right? We want to know where I used Logger::LogWarn.

File: src/logutil/logger.h [hunk 2]
```diff
@@ -55,6 +55,8 @@ class Logger
   static void LogInfo(const string& message);
   static void LogWarn(const string& message);
   static void LogError(const string& message);
+  static string s;
```
Comment (Dante-666) [inline src/logutil/logger.h:58]: This is not required, and static objects must be initialized. That's why you are getting errors in the build.

File: src/logutil/logger.h [hunk 3]
```diff
@@ -55,6 +55,8 @@ class Logger
...
+  friend class LoggerImpl;
```
Comment (Dante-666) [inline src/logutil/logger.h:59]: Again, the approach is not correct. Let's revisit the requirement again. I've wrote it in the PR.

**Review by Dante-666 — COMMENTED** [review]
General

**Comment by PracticalMetal** [comment]
Thanks for your reply, I will be working on this and fix all the issues thanks!

**Pull request closed without merging**